url9a01f43c15_poragchowdhury_securitygame_tgz-pJ8-securitygame_GameMasterJ8/src/securitygame/AttackerMonitor.java

```diff
@@ -0,0 +1,353 @@
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Random;
+
+/**
+ * Auxiliary class for creating and parsing .attack files.
+ * Attacker will use the three parameter constructor and combination of the two attack methods and the four probe methods
+ * to generate an .attack file. Have Attacker remember to call close() when finished for safety.
+ * Game Master will use the two parameter constructor for parsing original network and attacker file to generate the attacker's visible network and history.
+ *
+ * Actions deemed invalid will be charged the Parameters.INVALID_RATE value.
+ *
+ * @author      Marcus Gutierrez
+ * @version		2014/11/14
+ */
+
+public class AttackerMonitor
+{
+    private Network net;
+    private Network visibleNet;
+    
+    private String attackerName;
+    private String defenderName;
+    private String graphName;
+    private PrintWriter history;
+    
+    private int budget;
+    private int points;
+    private Random r;
+    private ArrayList<Node> availableNodes;
+
+    /**
+     * Constructor
+     * @param attackerName attacker name
+     * @param defenderName defender name
+     * @param graphName graph name
+     */
+    public AttackerMonitor(String attackerName, String defenderName, String graphName){
+        budget = Parameters.ATTACKER_BUDGET;
+        System.out.println("Initial Budget1: " + budget);
+        r = new Random();
+        this.attackerName = attackerName;
+        this.defenderName = defenderName;
+        this.graphName = graphName;
+        try {
+        net = Parser.parseGraph(defenderName + "-" + graphName + ".graph");
+        visibleNet = Parser.parseGraph(defenderName + "-" + graphName + "-hidden.graph");
+        availableNodes = visibleNet.getAvailableNodes();
+
+        //clears history and adds the public nodes to the history
+        history = new PrintWriter(new FileWriter(attackerName + "-" + defenderName + "-" + graphName + ".history", false));
+        ArrayList<Node> publicNodes = visibleNet.getCapturedNodes();
+        System.out.println("PUBLIC NODES: " + publicNodes.size());
+        for(int i = 0; i < publicNodes.size(); i++){
+            String publicString = "6," + publicNodes.get(i).getNodeID()+",";
+            ArrayList<Node> neighbors = publicNodes.get(i).getNeighborList();
+            for(int j = 0; j < neighbors.size(); j++)
+                publicString += neighbors.get(j).getNodeID() + ",";
+                publicString = publicString.substring(0, publicString.length() - 1);
+                history.println(publicString);
+            }
+            history.close();
+        } catch (IOException e) {e.printStackTrace();}
+        history.close();
+    }
+
+    /**
+     * Parses attacker's move from .attack file
+     */
+    public void readMove(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 93 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.